tasks/managed/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -384,22 +388,51 @@ spec:
             done <<< "$ARCHITECTURES"
 
             # Save component result to file
-            echo "$COMPONENT_JSON" > "$output_file"
+            echo "$COMPONENT_JSON" | tee "$output_file"
+        }
+
+        # Function to process a group of components with the same digest in series
+        process_digest_group() {
+            local digest=$1
+            local component_indices_string="$2"
+            
+            echo "Processing components with digest: $digest"
+            
+            # Process each component in the group in series
+            for component_index in $component_indices_string; do
+                echo "Processing component $component_index for digest $digest"
+                process_component "$component_index"
+            done
         }
 
         COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
 
+        # Group components by digest to prevent race conditions
+        declare -A digest_groups
+        for (( i=0; i < COMPONENTS; i++ )); do
+            CONTAINER_IMAGE=$(jq -r ".components[${i}].containerImage" "${SNAPSHOT_SPEC_FILE}")
+            DIGEST="${CONTAINER_IMAGE##*@}"
+            
+            if [[ -z "${digest_groups[$DIGEST]}" ]]; then
+                digest_groups[$DIGEST]="$i"
+            else
+                digest_groups[$DIGEST]="${digest_groups[$DIGEST]} $i"
+            fi
+        done
+
         RUNNING_JOBS="\j" # Bash parameter for number of jobs currently running
         SUCCESS=true
 
-        # Process each component with concurrency limit
-        for (( i=0; i < COMPONENTS; i++ )); do
-            echo "Starting component processing for component $i"
+        # Process each digest group in parallel, but components within each group in series
+        for digest in "${!digest_groups[@]}"; do
+            echo "Starting parallel processing for components with digest: $digest"
             # Limit batch size to concurrent limit
             while (( ${RUNNING_JOBS@P} >= $(params.concurrentLimit) )); do
                 wait -n || SUCCESS=false
             done
-            process_component $i > "/tmp/component_${i}.out" 2> "/tmp/component_${i}.err" &
+            process_digest_group "$digest" "${digest_groups[$digest]}" \
+              > "/tmp/component_group_${digest//[^a-zA-Z0-9]/_}.out" \
+              2> "/tmp/component_group_${digest//[^a-zA-Z0-9]/_}.err" &
```

Comment:
Just wondering what made you add the find&replace for the digest string. It will always be just something like `sha256:050072256b9a903bd914c0b2866828150cb229cea0efe5892e2b644d5dd3b34f`, no? So it should probably be fine as is? Or we could just change `:` to `-` if we're worried about that.